src/main/java/com/onlinebookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.onlinebookshop.service.impl;
+
+import com.onlinebookshop.dto.order.OrderItemResponseDto;
+import com.onlinebookshop.dto.order.OrderRequestDto;
+import com.onlinebookshop.dto.order.OrderResponseDto;
+import com.onlinebookshop.dto.order.UpdateOrderStatusDto;
+import com.onlinebookshop.mapper.OrderItemMapper;
+import com.onlinebookshop.mapper.OrderMapper;
+import com.onlinebookshop.model.Order;
+import com.onlinebookshop.model.OrderItem;
+import com.onlinebookshop.model.ShoppingCart;
+import com.onlinebookshop.model.User;
+import com.onlinebookshop.repository.order.OrderItemRepository;
+import com.onlinebookshop.repository.order.OrderRepository;
+import com.onlinebookshop.security.AuthenticationService;
+import com.onlinebookshop.service.OrderService;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public List<OrderResponseDto> getAll(Pageable pageable) {
+        return orderRepository.findOrderByUserId(authenticationService.getUserId()).stream()
+                .map(orderMapper::toResponseDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto save(OrderRequestDto requestDto) {
+        Order order = new Order();
+        User user = authenticationService.getCurrentUser();
+        ShoppingCart shoppingCart = authenticationService.getCartOfCurrentUser();
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(shoppingCart.getCartItems().stream()
+                .map(cartItem -> cartItem.getBook()
+                        .getPrice()
+                        .multiply(BigDecimal.valueOf(cartItem.getQuantity())))
+                .reduce(BigDecimal.valueOf(0), BigDecimal::add));
+        order.setShippingAddress(requestDto.getShippingAddress());
+        Order savedOrder = orderRepository.save(order);
+        savedOrder.setOrderItems(shoppingCart.getCartItems().stream()
```

Comment:
Is it possible to save orderItems together with order entity? (check out cascades)

Reply:
Discussed in the private channel